### PR TITLE
Latency awareness

### PR DESCRIPTION
**Pull request by wprzytula**
This is the first attempt to introduce latency awareness. I've chosen to build it atop the current `DefaultPolicy`, because it is natural that LA simply adds yet another level of iterator permutation.

The PR includes identification-by-HostID PR. This is for identifying nodes in LatencyAwareness' HashMap. 

New things are in last 4 commits.

### CODE REVIEW

**Comment by wprzytula** [comment]
@havaker @cvybhu Waiting for opinions on design adopted.

File: scylla/src/transport/load_balancing/default.rs [hunk 1]
```diff
@@ -244,8 +244,9 @@ impl DefaultPolicy {
         vec.into_iter()
     }
 
-    fn is_alive(node: &NodeRef<'_>) -> bool {
-        node.is_enabled() && !node.is_down()
+    fn is_alive(_node: &NodeRef<'_>) -> bool {
+        true // for now, we leave this as stub, until we have time to improve node events.
+        // node.is_enabled() && !node.is_down()
```
Comment (cvybhu) [inline scylla/src/transport/load_balancing/default.rs]: is_enabled

File: scylla/src/transport/load_balancing/default.rs [hunk 2]
```diff
@@ -9,33 +11,49 @@ use rand::{
...
 /// It can be configured to be datacenter-aware and token-aware.
 /// Datacenter failover for queries with non local consistency mode is also supported.
-#[derive(Debug)]
-pub struct DefaultPolicy {
+pub struct DefaultPolicy<Pred: Predicate> {
     preferred_datacenter: Option<String>,
     is_token_aware: bool,
     permit_dc_failover: bool,
+    latency_awareness: Option<LatencyAwareness>,
+    pick_predicate: Pred,
```
Comment (cvybhu) [inline scylla/src/transport/load_balancing/default.rs]: consider Box

**Comment by wprzytula** [comment]
Addressed both comments:
- Box'ed predicate instead of a generic
- is_enabled() stub for is_alive()

**Review by havaker — CHANGES_REQUESTED** [review]
Please drop this commit https://github.com/havaker/scylla-rust-driver/pull/6/commits/a11451367153f9651b05b02fea7160f54cd09f73

**Comment by wprzytula** [comment]
> Please drop this commit [a114513](https://github.com/havaker/scylla-rust-driver/commit/a11451367153f9651b05b02fea7160f54cd09f73)

Ok, but now `latency_aware.rs` is not deleted; I've made a mistake during recent rebase on main, so we probably need to remove it again. I would suggest removing it in a new commit, and I'll handle it in a final cleanup of this PR.

**Comment by havaker** [comment]
Please rebase to `havaker:lb2`.

**Comment by wprzytula** [comment]
> Please rebase to `havaker:lb2`.

done.

**Pull request closed without merging**